### PR TITLE
Per buffer contexts

### LINKED ISSUE #826: Security: proxy-mode is misleadingly global
Our proxy-mode is global, but as a mode it's meant to be buffer-local.
We are dangerously misleading our users here :(

The fix is to use per-buffer WebContexts.
The problem is that on doing so,

1. It generates hundreds of

```
(process:23271): libsoup-WARNING **: 11:40:55.141: Failed to execute query: table soup_hsts_policies already exists

(process:23271): libsoup-WARNING **: 11:40:55.141: Failed to execute query: database is locked
```

2. The browser tends to crash after about 1 minute, with

```
CORRUPTION WARNING in SBCL pid 23231(tid 0x7fd373667700):
Memory fault at (nil) (pc=0x7fd3796214e0, fp=0x7fd3640a4b48, sp=0x7fd3736661c0) tid 0x7fd373667700
The integrity of this image is possibly compromised.
Continuing with fingers crossed.

debugger invoked on a SB-SYS:MEMORY-FAULT-ERROR in thread
#<THREAD "cl-cffi-gtk main thread" RUNNING {1005E73CE3}>:
  Unhandled memory fault at #x0.

The current thread is not at the foreground,
SB-THREAD:RELEASE-FOREGROUND has to be called in #<SB-THREAD:THREAD "main thread" waiting on: #<MUTEX "thread result lock" owner: #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING {1005E73CE3}>> {1000650083}>
for this thread to enter the debugger.
```

The 2 issues could be related.  Let's get started with the first one.

### PR DESCRIPTION
**Pull request by Ambrevar**
Great news!  Since @aartaka implemented data-manager support in cl-webkit, we can now use per-buffer contexts without crashing Nyxt (I've ran it for half an hour so far).

This fixes #826.
Can you test on your site and confirm the browser does not crash (leave it up for at least 10 minutes).

@hsanjuan This might also fix the Protonmail logouts (#644).  Could you check if it works for you?  I've tried it here, Protonmail is still logged in after 30 minutes and many buffer switches.

### CODE REVIEW

**Comment by Ambrevar** [comment]
This could fix #717 as well. @khinsen Mind checking?

**Comment by Ambrevar** [comment]
This patch does not fix the libsoup warnings though, I wonder why.

File: source/browser.lisp [hunk 1]
```diff
@@ -946,10 +952,10 @@ current buffer."
             nil))))))
 
 (defmethod get-unique-window-identifier ((browser browser))
-  (incf (slot-value browser 'total-window-count)))
+  (format nil "~s" (incf (slot-value browser 'total-window-count))))
```
Comment (jmercouris) [inline source/browser.lisp:955]: Why format? You want a string specifically?

**Comment by Ambrevar** [comment]
Yes, our identifiers have always been strings!  (Look at the commit message.)

**Comment by khinsen** [comment]
> This could fix #717 as well. @khinsen Mind checking?

I will check, but not before August, sorry!

**Comment by hsanjuan** [comment]
I will need a couple of days, but my issue was already not happening in recent versions.

File: source/browser.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/browser.lisp:955]: Yes, our identifiers have always been strings!  (Look at the commit message.)

**Comment by hsanjuan** [comment]
I'm getting 

```
; caught ERROR:
;   READ error during COMPILE-FILE:
;   
;     Symbol "WEBKIT-WEBSITE-DATA-MANAGER" not found in the CL-WEBKIT2 package.
;   
;       Line: 421, Column: -1, File-Position: 17818
;   
;       Stream: #<SB-INT:FORM-TRACKING-STREAM for "file /home/hector/common-lisp/nyxt/source/renderer-gtk.lisp" {1007079673}>

; compilation aborted after 0:00:00.126
Unhandled UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                          {10009F0083}>:
  COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "nyxt/gtk" "renderer-gtk">
```

With both the commited version of submodules and the latest one.

**Comment by Ambrevar** [comment]
Does #853 fix it for you?
cl-webkit master has the webkit-website-data-manager symbol, it seems that your setup is not picking the right cl-webkit.  How are you building Nyxt?

**Comment by Ambrevar** [comment]
@hsanjuan Does it build now for you?

**Comment by hsanjuan** [comment]
I don't manage, I build with `make all`.

I am using latest cl-webkit from master and it fails after running `make clean`, with or without NYXT_INTERNAL_QUICKLISP=true... I am not sure it is using the right cl-webkit as I have it checked out? Could it be cached on an old version somewhere?

**Comment by hsanjuan** [comment]
Of course, the moment I sent the message above I realized I had `cl-webkit` in `~/common-lisp` because that was the thing before submodules were added. Removing that fixed the build.

**Comment by hsanjuan** [comment]
Very preliminary tests show no issues with Protonmail.

**Comment by Ambrevar** [comment]
Great! Thanks for confirming!

About the ~/common-lisp issue, I wonder if we can add a flag to tell ASDF to ignore this folder.  This is a recurring issue.  But now that we've got submodules, newcomers should not be hit by this again.

**Comment by Ambrevar** [comment]
Merged with acca0d81cee102830ff05c5c6fccbb0fc87bec46.  @khinsen Let us know when you can if this fixes #717 (anyways, we are not closing that issue just yet).

**Pull request closed without merging**